### PR TITLE
Add Conversion from Affine to PublicKey

### PR DESCRIPTION
**Pull request by Gauthamastro**
PR adds core::convert From trait for PublicKey to convert an Affine to PublicKey

### CODE REVIEW

**Review by athei — APPROVED** [review]
I see no harm in merging this. But @sorpaas should know.

**Review by sorpaas — CHANGES_REQUESTED** [review]
An affine may be invalid when converting to a public key. I'd prefer we use `TryFrom` instead.

**Comment by Gauthamastro** [comment]
@sorpaas I have implemented TryFrom, not really sure if my "if" condition is correct. Let me know

File: src/lib.rs [hunk 1]
```diff
@@ -303,11 +307,23 @@ impl Into<Affine> for PublicKey {
     }
 }
 
+impl TryFrom<Affine> for PublicKey {
+    type Error = Error;
+
+    fn try_from(value: Affine) -> Result<Self, Self::Error> {
+        if value.is_infinity() | !value.is_valid_var() {
```
Comment (sorpaas) [inline src/lib.rs:314]:
```suggestion
        if value.is_infinity() || !value.is_valid_var() {
```

**Comment by sorpaas** [comment]
Please also revert those formatting change. I'm not sure it's correct since the code should have been rustfmt-ed.

**Review by dvdplm — CHANGES_REQUESTED** [review]
There are many white space changes here that don't seem relevant to the PR. Do you mind reverting those?

**Comment by Gauthamastro** [comment]
@sorpaas @dvdplm I have fixed the formatting issue =)

**Review by dvdplm — APPROVED** [review]

**Review by sorpaas — APPROVED** [review]

**Pull request merged**